crates/driver/src/domain/competition/solution/settlement.rs

```diff
@@ -41,14 +51,16 @@ impl Settlement {
 
 /// A settlement which has been verified to be correct. In particular:
 ///
-/// - Simulation: the settlement has been simulated without reverting.
+/// - Simulation: the settlement has been simulated without reverting, including
+/// the case where no interactions were internalized.
 /// - Asset flow: the sum of tokens into and out of the settlement are
 /// non-negative, meaning that the solver doesn't take any tokens out of the
 /// settlement contract.
 /// - Internalization: internalized interactions only use trusted tokens.
 ///
 /// Such a settlement obeys the rules of the protocol and can be safely
-/// broadcast to the Ethereum network.
+/// broadcast to the Ethereum network without getting slashed. See the
+/// [`competition::Order::reward`] field for more about slashing.
```

Comment:
I'm not sure if we can prevent 100% of all slashable offenses with these tests. For example if a solver provides a solution that has a ridiculously low clearing price for some token and now other solver provides a solution we would probably still slash that solver since it would have been possible to achieve a better price. I believe something like this recently happened with `Quasilabs`.
Maybe a more defensive wording would be good.

Reply:
b3cd7eaececcb0596d36bc41b134b996260ff26c